core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCommonQueryBuilder.java

```diff
@@ -1891,9 +1895,7 @@ protected void appendGroupByClause(StringBuilder sbSelectFrom) {
         Set<String> clauses = new LinkedHashSet<String>();
         groupByManager.buildGroupByClauses(clauses);
 
-        int size = transformerGroups.size();
```

Comment:
* I consider this micro-optimization. As long as the performance of the loops is not proven to be a problem, I would go for readability vs micro-optimization.
* Apart from that I think that the for-each loop syntax considerably enhances readability as it removes 1-2 lines of nonsense code at every use-side.
* In this concrete instance I also don't understand why you store the `size` in a separate variable.
* Also, `transformerGroups` is of type `java.util.List`. So if, for some reason, say a `LinkedList` is used in the future instead of an `ArrayList`, the performance of the classical for-loop will suddenly suck (due to the lack of constant time random access) which might not be obvious to and may be overlooked by whoever introduces such a change.